src/IOCapture.jl

```diff
@@ -105,7 +105,13 @@ function capture(f; rethrow::Type=Any, color::Bool=false)
     # pipe to `output` in order to avoid the buffer filling up and stalling write() calls in
     # user code.
     output = IOBuffer()
-    buffer_redirect_task = @async write(output, pipe)
+    temp = IOBuffer()
+    buffer_redirect_task = @async begin
+        write(temp, pipe)
+        temp_data = take!(temp)
```

Comment:
Ideally, I'd like to avoid the allocation of `temp_data`. Can I `take!` from `temp` and then forward-rewind it so I can `take!` it a second time?

Reply:
output = IOBuffer()
    temp = IOBuffer()
    buffer_redirect_task = @async begin
        n_bytes = write(temp, pipe)
        seek(temp, 0)
        write(output, read(temp, n_bytes))
        seek(temp, 0)
        write(default_stdout, read(temp, n_bytes))
    end

might be a better alternative